realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpNetworkTransport.java

```diff
@@ -90,14 +128,87 @@ private synchronized OkHttpClient getClient(long timeoutMs) {
         return client;
     }
 
+    private synchronized OkHttpClient getStreamClient() {
+        if (streamClient == null) {
+            streamClient = new OkHttpClient.Builder()
+                    .readTimeout(0, TimeUnit.MILLISECONDS)
+                    .followRedirects(true)
+                    .addInterceptor(new LoggingInterceptor(httpLogObfuscator))
+                    .build();
+        }
+
+        return streamClient;
+    }
+
     // Parse Headers output from OKHttp to the format expected by ObjectStore
     private Map<String, String> parseHeaders(Headers headers) {
-        HashMap<String, String> osHeaders = new HashMap<>(headers.size()/2);
+        HashMap<String, String> osHeaders = new HashMap<>(headers.size() / 2);
         for (String key : headers.names()) {
             osHeaders.put(key, headers.get(key));
         }
         return osHeaders;
     }
 
+    public static class OkHttpResponse extends OsJavaNetworkTransport.Response {
```

Comment:
It feels a bit awkward and duplicates meaning to have this class named `OkHttpResponse` since it is already wrapped inside `OkHttpNetworkTransport`, so it becomes `OkHttpNetworkTransport.OkHttpResponse` ... `OkHttpNetworkTransport.Response` seems more readable.